src/doc/integrating_telio.md

```diff
@@ -1,5 +1,596 @@
 # Documentation For Integrating Telio Into Other Languages
 
-<!-- TODO: This page is a placeholder and will be populated by upcoming PR's when the actual specification will be more clear -->
+## Integration matrix
+
+| Feature/App                | TCLI | Telio-Apple | Android App | Windows App | Linux App | 
+|----------------------------|------|-------------|-------------|-------------|-----------|
+| Create/Destroy             | Yes | Yes | Yes |  | Yes |
+| Event Calback              | Yes | Yes | Yes |  | Yes |
+| Log Callback               | Yes | Yes | Yes |  | Yes |
+| Error Event Handling       | Yes | No | Yes |  | No |
+| Critical Recovery          | No | No | Yes |  | No |
+| Start/Stop Default Adapter | Yes | Yes | Yes |  | Yes |
+| Node Event Handling        | Yes | Yes | Yes |  | Yes |
+| VPN Client                 | Yes | Yes | Yes |  | Yes |
+| Meshnet Client             | Yes | Yes | Yes |  | Yes |
+| Magic DNS                  | Yes | Yes | Yes |  | Not via telio |
+| Meshnet + VPN              | Yes | Yes | Yes |  | Yes |
+| Meshnet + Exit Node        | Yes | Yes | Yes |  | Yes |
+| Error Handling             | Yes | Yes | Yes |  | Yes |
+
+## Integration feature list
+
+Note: the examples here are not using proper error handling for the sake of brevity. 
+
+### Create / Destroy
```

Comment:
I think addding global logger as a first example would make a lot of sence, also notting that it changed since v4.x